File: solutions/01array/11taelee.py
from typing import List

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        leftProduct = [1] * len(nums)
        rightProduct = [1] * len(nums)
        for i in range(1, len(nums)):
            leftProduct[i] = nums[i - 1] * leftProduct[i - 1]
        for j in range(len(nums) - 2, -1, -1):
            rightProduct[j] = rightProduct[j + 1] * nums[j + 1]
        return [leftProduct[i] * rightProduct[i] for i in range(len(nums)) ]


solution = Solution()
a = solution.productExceptSelf([1, 2, 3, 4])
print(a)

File: solutions/01array/11taelee.swift
class Solution {
    func productExceptSelf(_ nums: [Int]) -> [Int] {
        var leftProduct = Array(repeating: 1, count: nums.count)
        var rightProduct = Array(repeating: 1, count: nums.count)
        for i in 1..<nums.count {
            leftProduct[i] = leftProduct[i - 1] * nums[i - 1]
        }
        for j in (0...nums.count-2).reversed() {
            rightProduct[j] = rightProduct[j + 1] * nums[j + 1]
        }

        print(leftProduct)
        print(rightProduct)
        var resultArray: [Int] = []

        for i in 0..<nums.count {
            resultArray.append(leftProduct[i] * rightProduct[i])
        }
        return resultArray

    }
}

let solution: Solution = Solution()

print(solution.productExceptSelf([1, 2, 3, 4]))











